raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1895,12 +2021,14 @@ private Optional<Boolean> maybeHandleCommonResponse(
             }
         } else if (error == Errors.BROKER_NOT_AVAILABLE) {
             return Optional.of(false);
-        } else if (error == Errors.INCONSISTENT_GROUP_PROTOCOL) {
-            // For now we treat this as a fatal error. Once we have support for quorum
-            // reassignment, this error could suggest that either we or the recipient of
-            // the request just has stale voter information, which means we can retry
-            // after backing off.
-            throw new IllegalStateException("Received error indicating inconsistent voter sets");
```

Comment:
This error is never returned by KRaft.

Reply:
It's interesting that this was ever here -- seems that the error was created for consumer groups?

Reply:
Yeah, I was surprised to see this too. I think we were planning to implement voter reconfiguration as part of KIP-595. After some internal discussions we decided to pull it out of KIP-595 and spend more time designing reconfiguration.